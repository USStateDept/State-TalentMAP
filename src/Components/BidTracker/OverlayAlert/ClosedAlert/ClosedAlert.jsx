import PropTypes from 'prop-types';
import LinkButton from '../../../LinkButton';


const ClosedAlert = ({ title, date, bidIdUrl }, { condensedView }) => (
  <div className="bid-tracker-alert-container bid-tracker-alert-container--closed">
    <div className="top-text">
        The position {title} is no longer available.
    </div>
    {
      date &&
        <div className="date-text">
          {date}
        </div>
    }
    {
      condensedView &&
      <div className="usa-grid-full">
        <LinkButton toLink={bidIdUrl} className="tm-button-transparent">
          Go to Bid Tracker
        </LinkButton>
      </div>
    }
  </div>
);

ClosedAlert.propTypes = {
  title: PropTypes.string.isRequired,
  date: PropTypes.string,
  bidIdUrl: PropTypes.string,
};

ClosedAlert.defaultProps = {
  date: '',
  bidIdUrl: '',
};

ClosedAlert.contextTypes = {
  condensedView: PropTypes.bool,
};

export default ClosedAlert;
