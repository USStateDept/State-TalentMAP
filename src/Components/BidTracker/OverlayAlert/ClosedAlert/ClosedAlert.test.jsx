import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ClosedAlert from './ClosedAlert';

describe('ClosedAlertComponent', () => {
  it('is defined', () => {
    const wrapper = shallow(
      <ClosedAlert title="title" date="6.11.17" />,
    );
    expect(wrapper).toBeDefined();
  });

  it('matches snapshot', () => {
    const wrapper = shallow(
      <ClosedAlert title="title" date="6.11.17" />,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
