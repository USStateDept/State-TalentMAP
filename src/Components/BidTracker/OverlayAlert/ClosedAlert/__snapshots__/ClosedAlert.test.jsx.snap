// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClosedAlertComponent matches snapshot 1`] = `
<div
  className="bid-tracker-alert-container bid-tracker-alert-container--closed"
>
  <div
    className="top-text"
  >
    The position 
    title
     is no longer available.
  </div>
  <div
    className="date-text"
  >
    6.11.17
  </div>
</div>
`;
